src/bun.js/node-dns.exports.js

```diff
@@ -535,15 +548,21 @@ export const promises = {
     }
 
     resolve(hostname, rrtype) {
-      return dns.resolve(hostname, rrtype).then(promisifyResolve);
+      return dns.resolve(hostname, rrtype).then(promisifyResolve(rrtype));
```

Comment:
can we generate each of these functions at the top somewhere to avoid the extra function allocation

Reply:
done in https://github.com/oven-sh/bun/pull/2144/commits/9aeb64f0fe675a8382320842a5df8ba5f6e906a9